### PR TITLE
Refactoring/simplification

### PR DESCRIPTION
**Pull request by msmiths**
The changes in this pull request simplify the overall implementation of the mpOpenAPI feature.

### CODE REVIEW

**Comment by msmiths** [comment]
#run-libby-bot #build

**Comment by msmiths** [comment]
#run-libby-bot #build

**Comment by msmiths** [comment]
#run-libby-bot #build

File: dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/WebModuleOpenAPIProvider.java [hunk 1]
```diff
@@ -56,12 +75,68 @@ public String getOpenAPIDocument(Format format) {
...
+            /*
+             * We only store a single copy of the OpenAPI model for a given application, but the OpenAPI document could
+             * be requested by multiple clients concurrently.  As a result, we need to synchronize operations that
+             * update the model to ensure consistent behaviour. 
+             */
+            synchronized (openAPIModel) {
+                // Set the servers in the OpenAPI model and generate the OpenAPI document
+                openAPIModel.setServers(servers);
+                document = OpenAPIUtils.getOpenAPIDocument(openAPIModel, format);
+            }
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/WebModuleOpenAPIProvider.java:97]: This is much better.

However, is there a possibility that anything else could be reading `openAPIModel` at the same point that this modification is happening? (E.g. line 125)

I suspect that you need to synchronize around every access to `openAPIModel` to be safe here.

**Review by Azquelt — APPROVED** [review]

**Comment by msmiths** [comment]
#run-libby-bot #build

**Comment by msmiths** [comment]
#run-libby-bot

**Pull request merged**